doc/sphinx/addendum/canonical-structures.rst

```diff
@@ -219,6 +219,9 @@ The mixin component of the ``LEQ`` class contains all the extra content we
 are adding to ``EQ`` and ``LE``. In particular it contains the requirement
 that the two relations we are combining are compatible.
 
+The `class_of` projection of the `type` structure is annotated as *not canonical*;
+otherwise, bad things happen.
```

Comment:
What bad happens? I'm a bit lost ;-)

Reply:
See the warning at the end of that page of the manual: https://coq.github.io/doc/master/refman/addendum/canonical-structures.html

Reply:
I looked at the bug report and @robbertkrebbers does not say why he wants to disable CS on a projection. I've nothing against it, but (apart from not polluting indexes or the output of Print Cano..) I don't know why one would want to disable CS.

Reply:
Ah ok, here I'd say "it plays no role in the search of instances".

Reply:
What about the warning?

Reply:
Indeed, the warning should be documented. I find it hard to understand.

Reply:
> apart from not polluting indexes or the output of Print Cano..

That was indeed the main rationale.